apps/emqx_utils/src/emqx_utils_ets.erl

```diff
@@ -44,18 +59,18 @@ new(Tab, Opts) ->
             ok
     end.
 
+-spec lookup(ets:tab(), term()) -> [tuple()].
+lookup(Tab, Key) ->
+    ?safe_ets(ets:lookup(Tab, Key), []).
```

Comment:
Is it really necessary to add a try-catch block in every ets operation ? 
Maybe the cost too high? is it possible to only handle this during shutdown?

Reply:
I also thought about adding a generic `try catch` block directly in the cleanup but ultimately opted for this solution instead. 
As the goal is to avoid extra logging, it seems slightly risky to catch and any `badarg` error as it's quite generic (may be potentially triggered by other code besides ETS). 
As to the performance cost, I think `try catch` is cheap enough as long as stack trace is not inspected.

Reply:
IMO, I think this fix introduces a side effect for the general call, keeping the affected scope in the `clean up` would be better.
At the same time, I think try-catch is always expensive because it needs to save the calling environment for restoration when some exception happens

Reply:
try catch without attempting to retrieve the stacktrace should not impact performance in a observable way.
we can try to run some simple benchmark test to verify it.

Reply:
I agree with zhongwen and firest that hiding badarg error in a generic call path may hinder some potential issues especially during tests.

Reply:
Thanks for the feedback!
After additional discussion with @zmstone I've removed safe ETS wrappers and added `try .. catch` only in relevant shutdown functions that use async worker pool.